File: julia/trend/81b-violin.jl
using CSV, DataFrames, StatsPlots

# Read data from CSV file
df = CSV.read("series.csv", DataFrame)
rename!(df, Symbol.(strip.(string.(names(df)))))

# Extract the columns ys1, ys2, and ys3
data = [df.ys1, df.ys2, df.ys3]

# Create a violin plot using StatsPlots
violin(data, 
  labels = ["ys1", "ys2", "ys3"],
  linecolor = :black,
  legend = false,
  xlabel = "Variable",
  ylabel = "Value",
  title = "Violin Plot for ys1, ys2, and ys3",
  grid = false)

# Save plot as PNG
savefig("81b-violin.png")

File: python/trend/81b-violin.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Getting Matrix Values
pairCSV = np.genfromtxt("series.csv",
  skip_header=1, delimiter=",", dtype=float)

# Convert the data to a pandas DataFrame
cols_all = ['xs', 'ys1', 'ys2', 'ys3']
df = pd.DataFrame(pairCSV, columns=cols_all)

# Melt the DataFrame to long format for boxplot
df_melted = pd.melt(df,
  id_vars='xs', var_name='y', value_name='value')

# Create violinplot
plt.figure(figsize=(8, 6))
sns.violinplot(x='y', y='value', data=df_melted)

# Visualize the distribution of values
# across different categories.
plt.title('Violin Plot for ys1, ys2, and ys3')
plt.xlabel('Variable')
plt.ylabel('Value')
plt.show()
